EpikCord/client/http_client.py

```diff
@@ -0,0 +1,193 @@
+        """
+        Parameters
+        ----------
+        method : str
+            The method of the request.
+        url : str
+            The url of the request.
+        *args
+            The args to pass to the ClientSession.request method.
+        discord : bool
+            Whether this request is to Discord.
+        channel_id : int
+            The channel id of the request.
+        guild_id : int
+            The guild id of the request.
+        webhook_id : int
+            The webhook id of the request.
+        webhook_token : str
+            The webhook token of the request.
+        **kwargs
+            The kwargs to pass to the ClientSession.request method.
+
+        Returns
+        -------
+        Optional[aiohttp.ClientResponse]
+            The response of the request. Can be None if the request fails.
+        
+        Raises
+        ------
+        NotFound
+            If the request returns a 404 status code.
+        Forbidden
+            If the request returns a 403 status code.
+        Unauthorized
+            If the request returns a 401 status code.
+        BadRequest
+            If the request returns a 400 status code.
+        HTTPException
+            If the request returns a status code that is not OK and not any of the other exceptions.
+        """
+        if not discord:
+            return await self.session.request(method, url, *args, **kwargs)
+        if url.startswith("/"):
+            url = f"https://discord.com/api/v{self.version}{url}"
+        else:
+            url = f"https://discord.com/api/v{self.version}/{url}"
+
+        if url.endswith("/"):
+            url = url[:-1]
+        
+        bucket = self.buckets.get(url) or MockBucket()
+
+        await self.global_event.wait()
+        await bucket.wait()
+
+        async with self.session.request(method, url, *args, **kwargs) as response:
+            if isinstance(bucket, MockBucket) and response.headers.get("X-RateLimit-Bucket"):
+                if channel_id or guild_id or webhook_id or webhook_token:
+
+            if response.status == 429:
+                data = await response.json()
+                if data["global"]:
+                    self.global_event.clear()
+                bucket.clear()
+                await asyncio.sleep(data["retry_after"])
+                self.global_event.set()
+                bucket.set()
+                return await self.request(method, url, *args, **kwargs)
+
+            elif response.status == 404:
+                raise NotFound(response)
+            elif response.status == 403:
+                raise Forbidden(response)
+            elif response.status == 401:
+                raise Unauthorized(response)
+            elif response.status == 400:
+                raise BadRequest(response)
+            elif response.status != 200:
+                raise HTTPException(response)
```

Comment:
This could surely be simplified using a small hashmap

Reply:
Wait dont `response` have a `ok` attribute too?

Reply:
Yes afaik this was Github Copilot just shoving that in. This line 191 is incorrect. Will fix Soon:trademark: